README.md
# [Twitterwall2]( https://github.com/phasenraum2010/twitterwall2 )

+ Twitterwall with [spring:boot]( https://projects.spring.io/spring-boot/ ) for [heroku]( https://heroku.com )
+ More Info: https://phasenraum2010.github.io/twitterwall2/

## Development

### Repository:
- Get Source Code: git clone https://github.com/phasenraum2010/twitterwall2.git
- Project: https://github.com/phasenraum2010/twitterwall2/projects/1
- Issues: https://github.com/phasenraum2010/twitterwall2/issues

### heroku CLI
- heroku login
- heroku pg:psql --app {app-name}
- heroku logs -t --app {app-name}

### Setup Testing on Localhost
- Get Credentials from your Twitter-Account: [Refer to "Manage your Apps" on Twitter's dev pages](https://dev.twitter.com/apps)
- Edit your .profile with the Credentials from your Twitter-Account
  - export JDBC_DATABASE_URL=jdbc:postgresql://localhost:5432/postgres?user=postgres
  - export TWITTER_CONSUMER_KEY={Credentials from your Twitter-Account}
  - export TWITTER_CONSUMER_SECRET={Credentials from your Twitter-Account}
  - export TWITTER_ACCESS_TOKEN={Credentials from your Twitter-Account}
  - export TWITTER_ACCESS_TOKEN_SECRET={Credentials from your Twitter-Account}
- Edit app behaviour
  - export TWITTERWALL_FRONTEND_MAX_RESULTS=60
  - export TWITTERWALL_SHOW_USERS_MENU=true
  - export TWITTER_PAGE_SIZE=500
  - export TWITTER_FETCH_TESTDATA=true
  - export TWITTERWALL_WAIT_FOR_TEST=20000
  - export TWITTERWALL_URL_TEST_DATA_VERBOSE=true
  - export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_TWEETS=false
  - export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS=false
  - export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS_FROM_MENTION=true
  - export TWITTERWALL_SCHEDULER_ALLOW_SEARCH=true
  - export TWITTERWALL_SCHEDULER_HEROKU_DB_LIMIT=true
- Edit Use Case  
  - export TWITTER_SEARCH_TERM='#hibernate OR #java OR #TYPO3'
  - export TWITTERWALL_INFO_WEBPAGE=https://github.com/phasenraum2010/twitterwall2
  - export TWITTERWALL_THEME=typo3
  - export TWITTERWALL_APP_NAME='Twitterwall'
  - export TWITTERWALL_INFO_IMPRINT_SCREEN_NAME=port80guru
- run with: mvn clean spring-boot:run

## Contribute


### Fork, patch and contribute code

Feel free to fork Twitterwall [Git repository at GitHub][twitterwall-github] for your own use and
updates.

Contribute your fixes and new features back to the main codebase using
[GitHub pull requests][github-pull-req].

[twitterwall-github]: http://github.com/andriusvelykis/reflow-maven-skin/
[github-pull-req]: http://help.github.com/articles/using-pull-requests

### Issues (bug and feature tracker)

Please report any bugs found, feature requests or other issues on
[Twitterwall GitHub tracker][twitterwall-issues].

When creating a new issue, try following [necolas's guidelines][issue-guidelines].

[twitterwall-issues]: https://github.com/phasenraum2010/twitterwall2/issues
[issue-guidelines]: http://github.com/necolas/issue-guidelines/#readme

### Support

Let me know if you are using Twitterwall. I may get around to creating a showcase page listing user sites.

[Contact me][av-site] if you have questions about Twitterwall, or just like to say something about it.
If you _really really_ like it and want to support the author, I will be glad to
[accept a small donation][donate].

[av-site]: https://twitter.com/ThomasWoehlke
[donate]: https://www.paypal.me/ThomasWoehlke

### Author:  

**Thomas Wöhlke**
+ Twitter: https://twitter.com/ThomasWoehlke
+ Github: https://github.com/phasenraum2010
+ Blog: https://thomas-woehlke.blogspot.de

### Acknowledgements

+ Twitterwall Frontend is built on [Bootstrap][bootstrap].
+ These include icons from [Glyphicons][glyphicons] and web fonts from [Google][webfonts].
+ JavaScript goodies with [jQuery][jquery].
+ More icons by [fontawesome][fontawesome].
+ Twitterwall Backend is made with [Java][java] and [Spring-Boot][spring-boot].
+ Database [PostgreSQL][postgresql] and jdbc by [PostgreSQL-JDBC][postgresql-jdbc].
+ Running on Cloud-Service [Heroku][heroku].

[java]: http://www.oracle.com/technetwork/java/javase/downloads/index.html
[fontawesome]: http://fontawesome.io/icons/
[postgresql-jdbc]: https://jdbc.postgresql.org/
[postgresql]: https://www.postgresql.org/
[heroku]: https://heroku.com
[spring-boot]: https://projects.spring.io/spring-boot/
[bootstrap]: http://getbootstrap.com/
[glyphicons]: http://glyphicons.com/
[webfonts]: http://www.google.com/webfonts/
[jquery]: http://jquery.org

### Copyright and license

&copy; 2017 Thomas Wöhlke

Licensed under the GNU General Public License Version 2.0

http://www.gnu.org/licenses/old-licenses/gpl-2.0










docs/release-notes.md
---
layout: page
title: Release Notes
permalink: /release-notes/
---

The history of Twitterwall releases is documented below. For details of changes refer to the [project's GitHub issues][twitterwall-issues] or the [GitHub report][github-report].

[twitterwall-issues]: https://github.com/phasenraum2010/twitterwall2/issues?state=closed
[github-report]: github-report.html


## 1.0.0 / 2017-07-02

-   Initial release of Twitterwall.



env-config.sh
#!/opt/local/bin/bash

echo "------------------------------------------"
export TWITTERWALL_CONTEXT_TEST=true
export TWITTERWALL_FRONTEND_MAX_RESULTS=60
export TWITTER_PAGE_SIZE=500
export TWITTER_FETCH_TESTDATA=false
export TWITTERWALL_WAIT_FOR_TEST=120000
export TWITTERWALL_URL_TEST_DATA_VERBOSE=true
export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_TWEETS=false
export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS=false
export TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS_FROM_MENTION=true
export TWITTERWALL_SCHEDULER_ALLOW_SEARCH=true
export TWITTER_SEARCH_TERM='#hibernate OR #java OR #TYPO3'
export TWITTERWALL_INFO_WEBPAGE='https://github.com/phasenraum2010/twitterwall2'
export TWITTERWALL_THEME='typo3'
export TWITTERWALL_APP_NAME='Twitterwall'
export TWITTERWALL_INFO_IMPRINT_SCREEN_NAME='port80guru'
export TWITTERWALL_GOOGLE_ANALYTICS_ID='GOOGLE_ANALYTICS_ID'
export TWITTERWALL_SCHEDULER_HEROKU_DB_LIMIT=true
echo "------------------------------------------"
echo "TWITTERWALL_CONTEXT_TEST $TWITTERWALL_CONTEXT_TEST"
echo "TWITTERWALL_FRONTEND_MAX_RESULTS $TWITTERWALL_FRONTEND_MAX_RESULTS"
echo "TWITTER_PAGE_SIZE $TWITTER_PAGE_SIZE"
echo "TWITTER_FETCH_TESTDATA $TWITTER_FETCH_TESTDATA"
echo "TWITTERWALL_WAIT_FOR_TEST $TWITTERWALL_WAIT_FOR_TEST"
echo "TWITTERWALL_URL_TEST_DATA_VERBOSE $TWITTERWALL_URL_TEST_DATA_VERBOSE"
echo "TWITTERWALL_SCHEDULER_ALLOW_UPDATE_TWEETS $TWITTERWALL_SCHEDULER_ALLOW_UPDATE_TWEETS"
echo "TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS $TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS"
echo "TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS_FROM_MENTION $TWITTERWALL_SCHEDULER_ALLOW_UPDATE_USERS_FROM_MENTION"
echo "TWITTERWALL_SCHEDULER_ALLOW_SEARCH $TWITTERWALL_SCHEDULER_ALLOW_SEARCH"
echo "TWITTER_SEARCH_TERM $TWITTER_SEARCH_TERM"
echo "TWITTERWALL_INFO_WEBPAGE $TWITTERWALL_INFO_WEBPAGE"
echo "TWITTERWALL_THEME $TWITTERWALL_THEME"
echo "TWITTERWALL_APP_NAME $TWITTERWALL_APP_NAME"
echo "TWITTERWALL_INFO_IMPRINT_SCREEN_NAME $TWITTERWALL_INFO_IMPRINT_SCREEN_NAME"
echo "TWITTERWALL_GOOGLE_ANALYTICS_ID $TWITTERWALL_GOOGLE_ANALYTICS_ID"
echo "TWITTERWALL_SCHEDULER_HEROKU_DB_LIMIT $TWITTERWALL_SCHEDULER_HEROKU_DB_LIMIT"
echo "------------------------------------------"
